packages/fxa-shared/subscriptions/types.ts

```diff
@@ -14,7 +19,26 @@ export interface RawMetadata {
 }
 
 // A mapping of OAuth client ids to their corresponding capabilities.
-export type ClientIdCapabilityMap = Record<string, string[]>;
+export type ClientIdCapabilityMap = Record<string, readonly string[]>;
+export const ClientIdCapabilityMap = {
+  merge(
+    a: ClientIdCapabilityMap,
+    b: ClientIdCapabilityMap
+  ): ClientIdCapabilityMap {
+    return Object.entries(b).reduce((acc, [clientId, capabilities]) => {
+      if (!acc[clientId]) {
+        acc[clientId] = cloneDeep(capabilities);
+        return acc;
+      }
+      for (const capability of capabilities) {
+        if (!acc[clientId].includes(capability)) {
+          acc[clientId] = [...acc[clientId], capability];
+        }
+      }
+      return acc;
+    }, cloneDeep(a));
+  },
+};
```

Comment:
This probably shouldn't be located within a `types` file.

Reply:
Having a single export point for this pattern is key to utilizing the ability to have both a type and concrete value of the same name. Of course, if we can't have functions in files named `types`, then perhaps we can't have types anywhere but types files? This would be a bit problematic, and we likely need better naming schemes and organization.